AWSPinpoint/AWSPinpointEndpointProfile.m

```diff
@@ -121,12 +126,32 @@ - (BOOL) isApplicationLevelOptOut:(AWSPinpointContext *) context {
     return NO;
 }
 
+- (void) setOptOut:(NSString *)optOut {
+    NSString *overrideDefaultOptOut = [self.userDefaults stringForKey:AWSPinpointOverrideDefaultOptOutKey];
+    if (![overrideDefaultOptOut isEqualToString:optOut]) {
+        [self.userDefaults setObject:optOut forKey:AWSPinpointOverrideDefaultOptOutKey];
+        [self.userDefaults synchronize];
+    }
+
+    _optOut = optOut;
+}
+- (NSString *)optOut {
+    return _optOut;
+}
```

Comment:
It is generally advisable not to override setters and getters for properties.  I think it would be better to introduce a public API which allows them to override the optValue  perhaps something like
```
- (void) overrideOptOut:(NSString *)optOut {
        self.optOut = optOut;
        [self.userDefaults setObject:optOut forKey:AWSPinpointOverrideDefaultOptOutKey];
        [self.userDefaults synchronize];
}
```

Reply:
resolving tihs convo- we chatted about this and decided to keep the custom setting 
- developers are already using this in their code, but SDK was not respecting this. those facing the bug will be fixed if they upgrade to the latest SDK once this is released
- Android's optOut is read-only, platform parity means Android's pinpoint sdk will also need to make this field writable.